source/code/plugins/patch_management_lib.rb

```diff
@@ -12,28 +12,48 @@ class LinuxUpdates
     MAJOR_MINOR_VERSION_REGEX = /([^\.]+)\.([^\.]+).*/
-
-    @@agentDetailsMap = Hash[(`cat /etc/opt/microsoft/omsagent/conf/omsadmin.conf`.split(/\r?\n/).map {|s| 
-                                s.split("=")}).map {|key, value| 
-                                    [key, value]
-                                }
-                            ]
-
-    @@hostOSDetailsMap = Hash[(`cat /etc/opt/microsoft/scx/conf/scx-release`.split(/\r?\n/).map {|s|
-                               s.split("=")}).map {|key, value|
-                                    [key, value]
-                                }
-                            ]
+    OMS_ADMIN_FILE = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
+    SCX_RELEASE_FILE = "/etc/opt/microsoft/scx/conf/scx-release"
+
+    @@agentDetailsMap = if File.exist?(OMS_ADMIN_FILE)
+                            Hash[ File.open(OMS_ADMIN_FILE, "r") do |f|
+                                    f.each_line do |line|
+                                        line.split(/\r?\n/).map {|s| s.split("=")}.map {|key, value| [key, value]}
+                                    end
+                                end ]
+                        end
+
+    @@hostOSDetailsMap = if File.exist?(SCX_RELEASE_FILE)
+                            Hash[ File.open(SCX_RELEASE_FILE, "r") do |f|
+                                    f.each_line do |line|
+                                        line.split(/\r?\n/).map {|s| s.split("=")}.map {|key, value| [key, value]}
+                                    end
+                                end ]
+                        end
+
+    # @@hostOSDetailsMap = Hash[(`cat /etc/opt/microsoft/scx/conf/scx-release`.split(/\r?\n/).map {|s|
+    #                            s.split("=")}).map {|key, value|
+    #                                 [key, value]
+    #                             }
+    #                         ]
     
-    @os_major_version = @@hostOSDetailsMap["OSVersion"][MAJOR_MINOR_VERSION_REGEX, 1] unless @@hostOSDetailsMap["OSVersion"].nil?
-    @os_minor_version = @@hostOSDetailsMap["OSVersion"][MAJOR_MINOR_VERSION_REGEX, 2] unless @@hostOSDetailsMap["OSVersion"].nil?
-
     # This temporary fix is for version management for cache lookup.                        
-    def self.getOSShortName()
-        # match string of the form (1 or more non . chars)- followed by a . - (1 or more non . chars) - followed by anything
+    def self.getOSShortName(os_short_name = nil, os_version=nil)
         version = ""
 
-        case @@hostOSDetailsMap["OSShortName"].split("_")[0]
+        # match string of the form (1 or more non . chars)- followed by a . - (1 or more non . chars) - followed by anything
+        osName = (os_short_name == nil) ? @@hostOSDetailsMap["OSShortName"].split("_")[0] : os_short_name.split("_")[0]
+        if (os_version == nil)
+            @os_major_version = @@hostOSDetailsMap["OSVersion"][MAJOR_MINOR_VERSION_REGEX, 1] unless @@hostOSDetailsMap["OSVersion"].nil?
```

Comment:
os_version is being passed to the function. If it does not get anything, then it'll try to find the file (go and see if the file exists.) The version setting happens in the later parts where some flooring is happening.